### PR TITLE
Fix renovate support for `xgo`

### PR DESCRIPTION
**Pull request by pat-s**
- Use semver versions instead of `.x` (not detectable by renovate)
- Include xgo version in Makefile
- Clean `Dockerfile.make`: don't pin alpine versions, simplify node tags

### CODE REVIEW

File: .cspell.json [hunk 1]
```diff
@@ -129,6 +129,7 @@
     "nosniff",
     "ntfy",
     "octocat",
+    "ontainer",
```
Comment (6543) [inline .cspell.json:132]: why not just ignore .github/renovate.json

File: Makefile [hunk 2]
```diff
@@ -39,7 +39,7 @@ CGO_ENABLED ?= 1 # only used to compile server
 
 HAS_GO = $(shell hash go > /dev/null 2>&1 && echo "GO" || echo "NOGO" )
 ifeq ($(HAS_GO),GO)
-	XGO_VERSION ?= go-1.20.x
+	XGO_VERSION ?= 1.22.8
```
Comment (xoxys) [inline Makefile:42]: Sure that this change still works?

**Review by xoxys — COMMENTED** [review]
xgo support would have been possible with the `x` version as well https://github.com/thegeeklab/renovate-presets/blob/main/golang.json#L40-L45

File: Makefile (hunk 2 above)
Comment (xoxys) [reply Makefile:42]: I don't think so, see https://pkg.go.dev/src.techknowlogick.com/xgo#readme-go-releases

**Comment by pat-s** [comment]
Ah, then we should change it to this 👍

**Comment by pat-s** [comment]
Switched to @xoxys approach. Linked PR in preset: https://github.com/woodpecker-ci/renovate-config/pull/21

File: .github/renovate.json [hunk 3]
```diff
@@ -30,7 +31,7 @@
     },
     {
       "groupName": "golang-lang",
-      "matchPackagePatterns": ["^golang$", "xgo"],
```
Comment (xoxys) [inline .github/renovate.json:33]: Why have you removed "xgo" from the group here? IMO xgo and golang updates should still be grouped.

Comment (pat-s) [reply .github/renovate.json]: Ah, not on purpose. Thanks for catching!

**Review by xoxys — APPROVED** [review]

**Pull request merged**